action_server/src/sema4ai/action_server/_new_project_helpers.py

```diff
@@ -0,0 +1,110 @@
+import io
+import datetime
+import logging
+import os
+import zipfile
+import yaml
+import requests
+from pathlib import Path
+from typing import Dict, List, Union, Optional
+from pydantic.main import BaseModel
+
+from ._settings import get_default_settings_dir
+
+TEMPLATES_METADATA_URL = "https://downloads.robocorp.com/action-templates/action-templates.yaml"
+TEMPLATES_PACKAGE_URL = "https://downloads.robocorp.com/action-templates/action-templates.zip"
+
+ACTION_TEMPLATES_METADATA_FILENAME = "action-templates.yaml"
+
+ACTION_TEMPLATES_DIR = Path(f"{get_default_settings_dir()}/action-templates")
+ACTION_TEMPLATES_METADATA_PATH = Path(f"{ACTION_TEMPLATES_DIR}/{ACTION_TEMPLATES_METADATA_FILENAME}")
+
+log = logging.getLogger(__name__)
+
+
+class ActionTemplate(BaseModel):
+    name: str
+    description: str
+
+
+class ActionTemplatesMetadata(BaseModel):
+    hash: str
+    url: str
+    date: datetime.datetime
+    templates: List[ActionTemplate]
+
+
+def _ensure_latest_templates():
+    # Ensures the existence of the latest templates package.
+    # It downloads the latest templates metadata file, and compares the hash with
+    # the metadata hold locally (if exists).
+    # If there is no match (or metadata is not available locally), it will download
+    # the templates package.
+
+    os.makedirs(ACTION_TEMPLATES_DIR, exist_ok=True)
+
+    local_metadata = _get_local_templates_metadata()
+
+    new_metadata_content = requests.get(TEMPLATES_METADATA_URL).text
```

Comment:
As accessing the network is common to fail, this should probably show a nicer message when it fails -- also, if templates were already in place, maybe it can just proceed with a log message and use what's already downloaded.

Reply:
I decided to leave error handling to the caller (`_new_project.py` -> `create_new_project`) - let me know if you think that's okay, or we should add more logs closer to the place the error is raised.
In any case, if `_ensure_latest_templates` fails at any point, a message will be shown to the user, but the main command will continue, using already cached templates (if available).